apps/system/js/tracking_notice.js

```diff
@@ -0,0 +1,152 @@
+
+  TrackingNotice.prototype.customID = 'tracking-notice';
+
+  TrackingNotice.prototype.DEBUG = false;
+
+  TrackingNotice.prototype.name = 'TrackingNotice';
+
+  TrackingNotice.prototype.containerElement =
+    document.getElementById('fullscreen-overlay');
+
+  TrackingNotice.prototype.renderDialog = function() {
+    LazyLoader.load('shared/elements/gaia_switch/script.js', function() {
+      this.render();
+      this.boundShowIfNeeded = this.showIfNeeded.bind(this);
+      window.addEventListener('appopened', this.boundShowIfNeeded);
+    }.bind(this));
+  };
+
+  TrackingNotice.prototype.view = function view() {
+    return `<section id="tracking-notice" data-z-index-level="system-dialog"
+              class='hidden dialog'>
+              <div class="panel">
+                <header><h1 data-l10n-id="tracking-notice-header"></h1></header>
+                <p id="tracking-notice-copy"
+                data-l10n-id="tracking-notice-description"></p>
+                <img id="tracking-notice-image"></img>
+                <span data-l10n-id="tracking-protection"></span>
+                <div class="setting">
+                  <label data-l10n-id="tracking-protection"
+                  for="tracking-protection-toggle"></label>
+                  <gaia-switch id="tracking-protection-toggle">
+                  </gaia-switch>
+                  <p data-l10n-id="tracking-protection-setting-description"
+                  class="subtitle"></p>
+                </div>
+              </div>
+              <div class="confirm" id="tracking-notice-confirm">
+                <p data-l10n-id="tracking-notice-confirm"></p>
+              </div>
+            </section>`;
+  };
+
+
+  TrackingNotice.prototype._registerEvents = function() {
+    var self = this;
+    self.confirm.addEventListener('click', function onConfirm() {
+      self.confirm.removeEventListener('click', onConfirm);
+      self.confirmNotice();
+    });
+  };
+
+  TrackingNotice.prototype._fetchElements = function spl_initElements() {
+    this.element = document.querySelector('#' + this.instanceID);
+    this.confirm = this.element.querySelector('#tracking-notice-confirm');
+    this.setting = this.element.querySelector('gaia-switch');
+    this.description = this.element.querySelector('#tracking-notice-copy');
+    this.description.addEventListener('click', function() {
+      navigator.mozL10n.formatValue(TRACKING_NOTICE_LEARN_MORE_URL)
+      .then((url) => {
+        var activeApp = Service.query('AppWindowManager.getActiveWindow');
+        this.hide();
+        activeApp.navigate(url);
+        var element = activeApp.element;
+        // When the user clicks 'Learn More', go to that URL and display
+        // back the dialog when navigates to any other page.
+        element.addEventListener('_locationchange', function onLoadLearnMore() {
+          element.removeEventListener('_locationchange', onLoadLearnMore);
```

Comment:
This `removeEventListener` isn't working since you are listening with a bound function. You will either need to name the bound function with a `var onLoadLearnMore = function9) {..}.bind(this);` or save off `var self = this`.